packages/yoga/src/Checkbox/web/CheckMark.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { bool, string, func } from 'prop-types';
+import styled from 'styled-components';
+import { hexToRgb } from '@gympass/yoga-common';
+import { HiddenInput } from '../../shared';
+import check from './CheckIcon';
+
+const CheckMarkStyled = styled.div(
+  ({
+    checked,
+    disabled,
+    error,
+    theme: {
+      components: { checkbox },
+    },
+  }) => `
+    position: relative;
+    width: ${checkbox.width}px;
+    height: ${checkbox.height}px;
+    border-radius: ${checkbox.border.radii}px;
+    border-width: ${checkbox.border.width}px;
+    border-style: solid;
+
+    ${
+      checked
+        ? `background-color: ${checkbox.checked.backgroundColor}; 
+            background-image: url(${check}); 
+            background-size: 16px 16px;
+            background-repeat: no-repeat;
+            background-position: center;`
+        : ``
+    }
+
+    ${
+      disabled
+        ? `border-color: ${checkbox.disabled.border.color};`
+        : `border-color: ${checkbox.border.color};`
+    }
+
+    ${
+      disabled && checked
+        ? `background-color: ${checkbox.disabled.backgroundColor};`
+        : ``
+    }
+
+    ${error ? `border-color: ${checkbox.error.border.color};` : ``}
+
+    ${
+      error && checked
+        ? `background-color: ${checkbox.error.backgroundColor};`
+        : ``
+    }
+  `,
+);
+
+const CircleHover = styled.div(
```

Comment:
You don't need to make the same logic as you do in native. You could do it using `:before` here.